assets/src/components/detours/diversionPanel.tsx

```diff
@@ -57,7 +57,7 @@ export const DiversionPanel = ({
           </div>
         </section>
 
-        <section className="pb-3">
+        <section id="diversion-panel__directions" className="pb-3">
```

Comment:
Can we put a `title` prop on the section itself? Would that enable `getByTitle` to work?

Reply:
Would a `title` prop be better than `data-test-id`?

Reply:
fwiw, I try to avoid any of the other queries provided by RTL except in specific circumstances where it's unavoidable. I specifically try to avoid test id's and `getByTitle`
> `getByTitle`: The `title` attribute is not consistently read by screenreaders, and is not visible by default for sighted users
> 
> -- https://testing-library.com/docs/queries/about/#priority

I **_always_** reach for `byRole` first (double quoting for emphasis)
> _This should be your top preference for just about everything. There's not much you can't get with this (if you can't, it's possible your UI is inaccessible)._
>
> > `getByRole`: This can be used to query every element that is exposed in the [accessibility tree](https://developer.mozilla.org/en-US/docs/Glossary/AOM). With the `name` option you can filter the returned elements by their [accessible name](https://www.w3.org/TR/accname-1.1/). This should be your top preference for just about everything. There's not much you can't get with this (if you can't, it's possible your UI is inaccessible). Most often, this will be used with the `name` option like so: `getByRole('button', {name: /submit/i})`. Check the [list of roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques#Roles).
> >
> >-- https://testing-library.com/docs/queries/about/#priority

Reply:
A counterpoint to that option, you often mention that refactoring code should not result in failing tests. If all we wanted was an extra container to change the styling on the header, the `parentElement` approach would immediately break.

```
<h2 className="c-diversion-panel__h2">Detour Directions</h2>
```
=>
```
<div><h2 className="c-diversion-panel__h2">Detour Directions</h2></div>
```

You could get around this with a `.closest("section")`, but then if we changed `section` to say `article`, it would also fail.

Not necessarily trying to persuade you away from that, I don't mind updating tests in a case like this, but I want to make sure we have all the "facts" on the table.

Reply:
I'm super torn here. I don't think there's an answer that _fully_ works the way I want it to, and I'm still new enough to RTL that I don't have great instincts on this question. I get the sense that the two "best" (least bad?) approaches are:
 - Use a `data-test-id`, which will likely _work_, but leaks some test stuff into the code.
 - Use `.parentElement` or `.closest("section")`, which will mostly work, but does have the risk of non-user-facing DOM-adjustments breaking these tests

@mbta/skate-developers do you all have a preference between the two options above? (Or option 3: "other")?

Reply:
No preference here

Reply:
I went with `.closest("section")`. Hopefully that's not too brittle!

Reply:
[ Feedback: Remove id's from DiversionPanel in favor of more sophisticated selectors](https://github.com/mbta/skate/pull/2580/commits/83c46b58515f162614b887fdbeef33e697545372)